src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX = 30;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+
+        if (FigureType.values()[random.nextInt(FigureType.values().length)].name()
+                .equals(FigureType.CIRCLE.name())) {
+            return new Circle(colorSupplier.getRandomColor(), random.nextInt(MAX));
+        }
+        if (FigureType.values()[random.nextInt(FigureType.values().length)].name()
+                .equals(FigureType.SQUARE.name())) {
+            return new Square(colorSupplier.getRandomColor(), random.nextInt(MAX));
+        }
+        if (FigureType.values()[random.nextInt(FigureType.values().length)].name()
+                .equals(FigureType.RECTANGLE.name())) {
+            return new Rectangle(colorSupplier.getRandomColor(),
+                    random.nextInt(MAX), random.nextInt(MAX));
+        }
+        if (FigureType.values()[random.nextInt(FigureType.values().length)].name()
+                .equals(FigureType.RIGHT_TRIANGLE.name())) {
+            return new RightTriangle(colorSupplier.getRandomColor(),
+                    random.nextInt(MAX), random.nextInt(MAX));
+        }
+        return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                    random.nextInt(MAX), random.nextInt(MAX), random.nextInt(MAX));
```

Comment:
Code should be refactored.

Introduce `random.nextInt(FigureType.values().length)` as local variable.
Introduce `(FigureType.values()[...].name()` as local variable.

For this proposes better to use `switch-case` expression.

Extract creating concrete figure to separate methods like `getRandomRightTriangle` and call it from `case` block.